### ISSUE TITLE
An empty `Color()` (all components set to 0.0) returns `true` in boolean context

### ISSUE THREAD

**Issue by Mickeon** [godotengine/godot#69009]
### Godot version

v4.0.beta.custom_build [1ce2bd63f]

### System information

Windows 10

### Issue description

In GDScript, a Color with all of its components set to 0.0 still returns **`true`** when forced in a boolean context. 

This has been documented to an extent in the class reference. Every Color, except specifically opaque black, evaluate to **`true`**. Do also note that the empty initialization of a Color (`Color()`) is the same as opaque black.
> Note: In a boolean context, **_a Color will evaluate to false if it's equal to Color(0, 0, 0, 1) (opaque black)_**. Otherwise, a Color will always evaluate to true.

The inconsistency lies in the way some other Variant types are handled:
- All Vectors (including **Vector4**) evaluate to **`false`** if all of their components are set to 0. 
- **String**s (and **StringName**s), **Dictionaries** and **Array**s evaluate to **`false`** if they are empty.

So, overall, I do still find this behavior to be a minor oddity worth pointing out. It makes a bit of sense if intentional, but when all components are 0.0 the colour is essentially empty. It is nothing.

### Steps to reproduce

Copy this snippet of GDScript code in your favourite method of choice.
```GDScript
	if (Color()):
		print("Opaque Black Color")

	if (Color.BLACK):
		print("Opaque Black Color")

	if (Color(0, 0, 0, 0)):
		print("All 0 Color printed")
```
Prints:
```
All 0 Color printed
```

### Minimal reproduction project

There's no need, it is very easy to reproduce.

**Comment by KoBeWi** [comment]
It's not that "zero" variants are false, just their default value (`String()`, `Array()`, `Signal()` etc.) and default value of most variants just happens to be 0. If zero color was false, it would be actually inconsistent. If anything, we can consider making it default `Color()`.

**Comment by Mickeon** [comment]
Not sure I understand the last sentence. `Color()` is already essentially the same a `Color.BLACK` and by extension also evaluates to false.

**Comment by KoBeWi** [comment]
I mean `Color()` could become `Color(0, 0, 0, 0)`.

**Comment by Mickeon** [comment]
That would make perfect sense to me. It almost goes back to the time I suggested the TRANSPARENT_BLACK constant.

**Comment by akien-mga** [comment]
This would break compat and require changing a *ton* of code which assumes that a default initialized color is opaque black. I don't see the point.

**Comment by mhilbrunner** [comment]
Yeah, heavily agree, at a max I'd document this oddity but it's so specific I don't even think thats really necessary.

The bigger thing for me is that this works at all and a Color is somehow a valid bool. That seems... weird?

**Comment by KoBeWi** [comment]
Every Variant is a valid bool.
